dash_extensions/enrich.py

```diff
@@ -612,22 +612,27 @@ def apply_serverside(self, callbacks):
             st_flex_key = callback.inputs.append(State(start_client_ctx, "data"))
             # Modify the callback function accordingly.
             f = callback.f
-            callback.f = skip_input_signal_add_output_signal(single_output, out_flex_key, in_flex_key, st_flex_key)(f)
+            callback.f = skip_input_signal_add_output_signal(num_outputs, out_flex_key, in_flex_key, st_flex_key)(f)
 
         return callbacks
 
 
-def skip_input_signal_add_output_signal(single_output, out_flex_key, in_flex_key, st_flex_key):
+def skip_input_signal_add_output_signal(num_outputs, out_flex_key, in_flex_key, st_flex_key):
     def wrapper(f):
         @functools.wraps(f)
         def decorated_function(*args, **kwargs):
             args, kwargs, fltr = _skip_inputs(args, kwargs, [in_flex_key, st_flex_key])
             cached_ctx = fltr[1]
+            single_output = num_outputs <= 1
             if cached_ctx is not None and "triggered" in cached_ctx:
                 ctx = context_value.get()
                 ctx["triggered_inputs"] = cached_ctx["triggered"]
                 context_value.set(ctx)
-            outputs = f(*args, **kwargs)
+            try:
+                outputs = f(*args, **kwargs)
+            except Exception as e:
+                outputs = no_update if single_output else [no_update] * num_outputs       
```

Comment:
The current implementation "swallows" all exceptions, thereby making it harder for the developer to debug the code. I would suggest to add logging of the exception, i.e. something like,

```
            except Exception:
                logging.exception(f"Exception raised in blocking callback [{f.__name__}]")
```

Reply:
hey @emilhe , thanks for the feedback , fixed